src/basic/escape.c

```diff
@@ -504,6 +504,9 @@ char* shell_maybe_quote(const char *s, ShellEscapeFlags flags) {
          * string. Note that we treat benign UTF-8 characters as needing
          * escaping too, but that should be OK. */
 
+        if (FLAGS_SET(flags, SHELL_ESCAPE_EMPTY) && isempty(s))
+                return strdup("''"); /* We don't use $'' here in the POSIX mode. '' is fine too. */
```

Comment:
hmm, it's a bit weird to return `''` instead of `""` here, no? i mean, it sounds more systematic if we try to cut down on different escape modes, i.e. I'd really return `""` if posix mode is off here, and otherwise `$''`, but at the very least use `""` instead of `''`. but returning `""` if the user explicitly requested posix escaping via the flag sounds weird too

the commit msg is not inline with the used quotes here btw, it talks partially about `""` and in the title about `''`...

Reply:
I changed it to return "" now.